### PR TITLE
Add transaction resubmission

### PR DESCRIPTION
**Pull request by karlfloersch**

### CODE REVIEW

File: src/batch-submitter/batch-submitter.ts [hunk 1]
```diff
@@ -124,6 +126,49 @@ export abstract class BatchSubmitter {
...
+    )]
+    const val = await bPromise.any(promises)
+
+    if (val === 'resubmit') {
+      log.debug('Tx resubmission timeout reached. Resubmitting tx.')
+      const tx = {
+        to: response.to,
+        data: response.data,
+        nonce: response.nonce,
+        value: 0
```
Comment (karlfloersch) [inline src/batch-submitter/batch-submitter.ts:155]: Don't set value here

File: src/batch-submitter/batch-submitter.ts [hunk 2]
```diff
@@ -124,6 +126,49 @@ export abstract class BatchSubmitter {
...
+      }
```
Comment (karlfloersch) [inline src/batch-submitter/batch-submitter.ts:156]: In future version specify gas price

File: src/batch-submitter/batch-submitter.ts [hunk 3]
```diff
@@ -138,6 +140,50 @@ export abstract class BatchSubmitter {
...
+    signer: Signer,
+    numConfirmations: number,
+    resubmissionTimeout: number,
+    log: Logger,
+  ): Promise<TransactionReceipt> {
+    const receiptPromise = response.wait(numConfirmations)
+    receiptPromises.push(receiptPromise)
+    // Wait for the tx & if it takes too long resubmit
+    const sleepAndReturnResubmit = async (timeout: number) => {
+      await new Promise((r) => setTimeout(r, timeout))
```
Comment (ben-chain) [inline src/batch-submitter/batch-submitter.ts:155]:
```suggestion
      await new Promise((r) => setTimeout(r, timeout))
      log.debug(`Timeout exceeded for batch submission with nonce ${response.nonce}.  Resolving to resubmit...`)
```

File: src/batch-submitter/batch-submitter.ts [hunk 4]
```diff
@@ -138,6 +140,50 @@ export abstract class BatchSubmitter {
...
+      return 'resubmit'
+    }
+    const promises = [...receiptPromises, sleepAndReturnResubmit(
+      resubmissionTimeout
+    )]
+    const val = await bPromise.any(promises)
+
+    if (val === 'resubmit') {
+      log.debug('Tx resubmission timeout reached. Resubmitting tx.')
```
Comment (ben-chain) [inline src/batch-submitter/batch-submitter.ts:164]:
```suggestion
      log.debug(`Tx resubmission timeout promise for nonce ${response.nonce} resolved, resubmitting tx...`)
```

**Review by ben-chain — DISMISSED** [review]
LGTM, left a couple logging improvements

**Review by K-Ho — APPROVED** [review]
Lgtm

**Pull request merged**